addons/mrp/models/mrp_production.py

```diff
@@ -854,17 +854,17 @@ def _get_move_finished_values(self, product_id, product_uom_qty, product_uom, op
         if len(group_orders) > 1:
             move_dest_ids |= group_orders[0].move_finished_ids.filtered(lambda m: m.product_id == self.product_id).move_dest_ids
         date_planned_finished = self.date_planned_start + relativedelta(days=self.product_id.produce_delay)
-        date_planned_finished = date_planned_finished + relativedelta(days=self.company_id.manufacturing_lead)
-        if date_planned_finished == self.date_planned_start:
-            date_planned_finished = date_planned_finished + relativedelta(hours=1)
+        self.date_planned_finished = date_planned_finished + relativedelta(days=self.company_id.manufacturing_lead)
```

Comment:
I agree on the analysis but I see one problem with the implementation: You should adjust MO date_planned_finished before reaching this point, this method is a hook for adjusting the values of the stock move and it is not a suitable place to change any MO field.

Reply:
That's right, I already changed it. What do you think now?

Reply:
Looks better now, thanks!